src/PrestaShopBundle/Form/Admin/Catalog/Category/AbstractCategoryType.php

```diff
@@ -67,51 +70,80 @@ abstract class AbstractCategoryType extends TranslatorAwareType
      */
     private $configuration;
 
+    /**
+     * @var Router
+     */
+    private $router;
+
     /**
      * @param TranslatorInterface $translator
      * @param array $locales
      * @param array $customerGroupChoices
      * @param FeatureInterface $multistoreFeature
      * @param ConfigurationInterface $configuration
+     * @param Router $router
      */
     public function __construct(
         TranslatorInterface $translator,
         array $locales,
         array $customerGroupChoices,
         FeatureInterface $multistoreFeature,
-        ConfigurationInterface $configuration
+        ConfigurationInterface $configuration,
+        Router $router
     ) {
         parent::__construct($translator, $locales);
 
         $this->customerGroupChoices = $customerGroupChoices;
         $this->multistoreFeature = $multistoreFeature;
         $this->configuration = $configuration;
+        $this->router = $router;
     }
 
     /**
      * {@inheritdoc}
      */
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
+        $disableMenuThumbnailsUpload = false;
+        if (isset($options['id_category'])) {
+            $disableMenuThumbnailsUpload = $options['disable_menu_thumbnails_upload'];
+        }
+        $genericCharactersHint = $this->trans('Invalid characters:', 'Admin.Global') . ' <>;=#{}';
+
         $builder
             ->add('name', TranslatableType::class, [
+                'label' => $this->trans('Name', 'Admin.Global'),
+                'help' => $genericCharactersHint,
                 'type' => TextType::class,
                 'constraints' => [
                     new DefaultLanguage(),
                 ],
                 'options' => [
+                    'attr' => [
+                        'maxlength' => static::MAX_TITLE_LENGTH,
+                    ],
```

Comment:
I can't comment the lines below but I think you could replace most of the regexp constraints in this form type with `TypedRegex`

Reply:
Yeah you are right, replaced with TypedRegex